### DISCUSSION TITLE
Стоимость с учетом страховки

### DISCUSSION THREAD

**Discussion by Scharbhen** [cdek-it/widget#24] [Q&A] [answered]
Добрый день!

Можно ли как-то настроить виджет для расчета стоимости доставки с учетом страховки? 

Пробовал передавать ему в инициализацию блок типа такого, но расчет все-равно происходит без учета страховки. 
services:[
        {
          code: "INSURANCE",
          parameter: "20000"
        },
      ],

**Comment by vermorag** [comment, marked as answer]
Добрый день. В настоящий момент, такого функционала в виджете нет. Задача на доработку уже существует, стоит в очереди на 2Q.2024.

**Reply by vermorag** [reply]
Номер вашей задачи на доработку CMS-593. 
Если требуется, можете подключиться к рассылке по релизу всех новых версий наших плагинов. Письма в рассылке о новых версиях содержат номера задач команды разработки, входящих в патч, а также краткое описание обновлений. 
Для подписки на рассылку напишите, пожалуйста, в телеграм: https://t.me/cdek_it_bot , раздел Интеграция для интернет-магазинов.

**Reply by neverfan** [reply]
Добрый день, подскажите какой текущий статус задачи? 

Очень-очень нужен данный функционал, иначе клиентам сообщаем неправильную стоимость доставки.

Если это сложно/долго реализовать, то нужна хотя бы возможность скрывать стоимость посчитанной доставки, чтобы посчитать ее отдельным запросом к cdek api через backend, т.к. там можно указать параметры страховки.

upd: прошу прощения, нашел в faq как отключить вывод стоимости (странно, что это не описано в настройках), но получается, что реализацию расчета полной стоимости (/v2/calculator/tariff) нужно реализовывать отдельно и самостоятельно через cdek sdk, хотя вроде бы в виджете все для этого есть.

**Comment by piton-zel** [comment]
Можно еще как решение, повесится на onCalculate, например вот так я добавляю 20%, т.к. виджет не учитывает что компания с НДС
```
params['onCalculate'] = function (prices) {
                let f;
                if (prices.office) for (f = prices.office.length - 1; f >= 0; --f) prices.office[f].delivery_sum *= 1.2;
                if (prices.door) for (f = prices.door.length - 1; f >= 0; --f) prices.door[f].delivery_sum *= 1.2;
                if (prices.pickup) for (f = prices.pickup.length - 1; f >= 0; --f) prices.pickup[f].delivery_sum *= 1.2;
};
```
Думаю что эта история не задумывалась в разработке, но как обходной вариант работает
